### DISCUSSION TITLE
Defining complex initial conditions

### DISCUSSION THREAD

**Discussion by Mide2001** [lululxvi/deepxde#1229] [Q&A]
I want to evaluate PINNs and compare them to numerical methods for engineering applications and one of these applications is the 2D heat equation. So I'm trying to set up a complex initial condition(IC) over a square domain. The IC that I want to set up is an annulus. I'm able to do this with basic code but I can't figure out how to do this with DeepXDE. I want to be able to define the IC, in this case, an annulus, and the magnitude at each point in this case the temperature. Is there a way to do this?

Here's the code for the basic solution:

> for i in range(nx):
 for j in range(ny):
  if ( ( (i*dx-1)**2+(j*dy-1)**2 <= 0.3)& ((i*dx-1)**2+(j*dy-1)**2>=.1) ):
   u[i,j] = 2.1
  else:
    u[i,j] = 1
![image](https://user-images.githubusercontent.com/88860159/229376898-bff4052c-5a46-46a7-b587-141255777f0d.png)

And here's what i have in DeepXDE so far:
> def func(x):
    return -((((x[:, 0:1]-2.5)**2)+((x[:, 1:2]-2.5)**2)) -0.3)
![image](https://user-images.githubusercontent.com/88860159/229376860-7fe4dc53-f4be-41c9-a71a-c4b09ea1cb56.png)

**Comment by lululxvi** [comment]
See FAQ "Q: More details and examples about initial conditions." and "Q: More details and examples about boundary conditions."

**Reply by Mide2001** [reply]
I've checked the FAQ's and unfortunately couldn't find any answers. Is there a way to assign a value to the initial conditions?

**Comment by lululxvi** [comment]
The `func` at https://deepxde.readthedocs.io/en/latest/modules/deepxde.icbc.html#deepxde.icbc.initial_conditions.IC can be any function, no matter how complex it is.
